crates/wasmtime/src/runtime/store.rs

```diff
@@ -2264,6 +2424,41 @@ at https://bytecodealliance.org/security.
 
         Ok(id)
     }
+
+    /// Set a pending exception. The `exnref` is taken and held on
+    /// this store to be fetched later by an unwind. This method does
+    /// *not* set up an unwind request on the TLS call state; that
+    /// must be done separately.
+    #[cfg(feature = "gc")]
+    pub(crate) fn set_pending_exception(&mut self, exnref: VMExnRef) {
+        debug_assert!(self.pending_exception.is_none());
+        self.pending_exception = Some(exnref);
+    }
+
+    /// Take a pending exception, if any.
+    #[cfg(feature = "gc")]
+    pub(crate) fn take_pending_exception(&mut self) -> Option<VMExnRef> {
+        self.pending_exception.take()
+    }
+
+    #[cfg(feature = "gc")]
+    fn take_pending_exception_rooted(&mut self) -> Option<Rooted<ExnRef>> {
+        self.take_pending_exception().map(|vmexnref| {
+            let mut nogc = AutoAssertNoGc::new(self);
+            let cloned = vmexnref.clone(nogc.require_gc_store_mut().unwrap());
+            ExnRef::from_cloned_gc_ref(&mut nogc, cloned.into())
```

Comment:
Is `clone` the right operation to call here? I would have expected we just thread through the `VMGcRef` as-is because we're effectively moving ownership into the return value